### ISSUE TITLE
Index disappears after compaction

### ISSUE THREAD

**Issue by e-dard** [influxdata/influxdb#8169]
While running a largish stress test using `influx_stress` on https://github.com/influxdata/influxdb/commit/5d67c424bfb44add94698e3be94da3c11c94d20c series data disappears after a while.

A reboot of the server bring the data back. It's probably index/snapshot related.

`influx_stress` config:

```
[provision]
  [provision.basic]
    enabled = true
    address = "localhost:8086"
    database = "stress"
    reset_database = true

[write]
  [write.point_generator]
    [write.point_generator.basic]
      enabled = true
      point_count = 100
      series_count = 400000
      tick = "10s"
      jitter = true
      measurement = "cpu"
      start_date = "2006-Jan-02"
      [[write.point_generator.basic.tag]]
        key = "host"
        value = "server"
      [[write.point_generator.basic.tag]]
        key = "location"
        value = "us-west"
      [[write.point_generator.basic.field]]
        key = "value"
        value = "float64"


  [write.influx_client]
    [write.influx_client.basic]
      enabled = true
      addresses = ["localhost:8086"] # stress_test_server runs on port 1234
      database = "stress"
      precision = "n"
      batch_size = 10000
      batch_interval = "0s"
      concurrency = 100
      ssl = false
      format = "line_http" # line_udp, graphite_tcp, graphite_udp

[read]
  [read.query_generator]
    [read.query_generator.basic]
      template = "SELECT count(value) FROM cpu where host='server-%v'"
      query_count = 700

  [read.query_client]
    [read.query_client.basic]
      enabled = true
      addresses = ["localhost:8086"]
      database = "stress"
      query_interval = "100ms"
      concurrency = 10

```

**Comment by jwilder** [comment]
This is actually an issue w/ the max values per tag limit and some re-slicing occurring on the wrong line.

**Comment by jwilder** [comment]
Fixed via #8172
